cli/src/fetch-recipe.ts

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import fetch from 'node-fetch';
+import { promises as fs } from 'fs';
+
+import { Recipe } from './recipe';
+import STATUS_CODES from './status-codes';
+
+type RecipeFile = {
+  type: 'file';
+  name: string;
+  download_url: string;
+};
+
+type RecipeDir = {
+  type: 'dir';
+  name: string;
+  path: string;
+};
+
+type RecipeNode = RecipeFile | RecipeDir;
+
+const COOKBOOK_BASE_URL =
+  'https://api.github.com/repos/Wolox/frontend-cookbook/contents';
+
+const COOKBOOK_FOLDERS_EXCEPTIONS: Record<string, string> = {
+  react: 'cookbook-react/src/recipes'
+};
+const getCookbookUrl = (tech: string) =>
+  COOKBOOK_FOLDERS_EXCEPTIONS[tech] || `cookbook-${tech}/recipes`;
+
+const importFile = async (file: RecipeFile, currentDir = '') => {
+  // TODO: Support non text files
+  const response = await fetch(file.download_url, {
+    headers: { 'Content-Type': 'application/text' }
+  });
+
+  if (response.status === STATUS_CODES.NOT_FOUND) {
+    throw new Error(`The file with name '${file.name}' could not be found`);
+  } else if (
+    response.status < STATUS_CODES.OK ||
+    response.status > STATUS_CODES.MAX_CLIENT_ERROR
+  ) {
+    throw new Error(
+      `The following error has occurred requesting the file ${
+        file.name
+      }: \n ${response.json()}`
+    );
+  }
+
+  await fs.writeFile(`./${currentDir}/${file.name}`, await response.text());
+};
+
+export async function fetchRecipe(
+  recipe: Recipe,
+  currentDir?: string
+): Promise<void> {
+  if (currentDir) {
+    await fs.mkdir(`.${currentDir}`, { recursive: true });
+  }
+
+  const response = await fetch(
+    `${COOKBOOK_BASE_URL}/${getCookbookUrl(recipe.tech)}/${recipe.category}/${
+      recipe.name
+    }${currentDir || ''}`
+  );
+
+  if (response.status === STATUS_CODES.NOT_FOUND) {
```

Comment:
From line 67 to line 80 the code is the same as in `importFile`, line 37 (except for the first if's error message). maybe we can extract that